src/Service/Resolvers/Sql Query Structures/SqlInsertQueryStructure.cs

```diff
@@ -97,7 +99,9 @@ private void PopulateColumnsAndParams(string columnName, object? value)
         {
             InsertColumns.Add(columnName);
 
-            // If the column is referenced in the database policy, we remove it from the set.
+            // As we add columns to the InsertColumns list for SqlInsertQueryStructure one by one,
+            // we remove the columns (if present) from the FieldsReferencedInDbPolicyForCreateAction.
+            // This is necessary because any field referenced in database policy but not present in insert statement would result in an exception.
             FieldsReferencedInDbPolicyForCreateAction.Remove(columnName);
```

Comment:
Who populates this member `FieldsReferencedInDbPolicyForCreateAction`? I couldnt see a corresponding `.Add`

Reply:
https://github.com/Azure/data-api-builder/blob/d45c20e5fc0c317023444b88ed25a4f030dab5d1/src/Service/Parsers/ODataASTVisitor.cs#L68

Reply:
That code does it only for Create operation. Is there similar code for Update or Upsert operation?

Reply:
We want to add the backing columns only for create operation. Upsert consists of update+create. For the create part only, we will add the fields.

Reply:
Why for create part only? Why not for update part - dont we encounter that situation if its update policy?